### PR TITLE
Add e3sm_to_cmip

### PR DESCRIPTION
**Pull request by xylar**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/e3sm_to_cmip/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    and Tables directories. Test holds the common_user_input.json file which
+    can be used as a placeholder for the user supplied metadata, and Tables
+    holds all the CMIP6 variable tables.
+  doc_url: https://github.com/E3SM-Project/e3sm_to_cmip
+  dev_url: https://github.com/E3SM-Project/e3sm_to_cmip
+
+extra:
+  recipe-maintainers:
+    - xylar
+    - sterlingbaldwin
```
Comment (xylar) [inline recipes/e3sm_to_cmip/meta.yaml]: @sterlingbaldwin, please acknowledge that you're willing to be a maintainer.

Comment (sterlingbaldwin) [reply recipes/e3sm_to_cmip/meta.yaml]: I so acknowledge.

**Comment by sterlingbaldwin** [comment]
Im happy to be a maintainer for this.

File: recipes/e3sm_to_cmip/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/E3SM-Project/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: a23d7440bab9950f6fef8a6e5fe60741e4e5be2a2421d37ca9a1739dab1deef1
+
+build:
+  skip: True  # [win or py<36]
```
Comment (xylar) [inline recipes/e3sm_to_cmip/meta.yaml]: @sterlingbaldwin, just to make sure you agree that we don't want to support Windows or python <3.6

File: recipes/e3sm_to_cmip/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    - xarray
+    - netcdf4
+    - dask
+    - scipy
+
+test:
+  commands:
+    - e3sm_to_cmip --help
+  imports:
+    - e3sm_to_cmip
```
Comment (xylar) [inline recipes/e3sm_to_cmip/meta.yaml]: @sterlingbaldwin, any other sanity checks you would add?

**Comment by xylar** [comment]
@conda-forge/help-python, would someone be willing to review this PR?

File: recipes/e3sm_to_cmip/meta.yaml (hunk 3 above)
Comment (xylar) [reply recipes/e3sm_to_cmip/meta.yaml]: Note: we can always add more checks later once the feedstock is built.

File: recipes/e3sm_to_cmip/meta.yaml (hunk 2 above)
Comment (marcelotrevisani) [reply recipes/e3sm_to_cmip/meta.yaml]: Is there anything specific for ``unix`` platform which cannot be run on ``windows``?

File: recipes/e3sm_to_cmip/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/E3SM-Project/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: a23d7440bab9950f6fef8a6e5fe60741e4e5be2a2421d37ca9a1739dab1deef1
+
+build:
+  skip: True  # [win or py<36]
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
```
Comment (marcelotrevisani) [inline recipes/e3sm_to_cmip/meta.yaml]: you may add the ``entry_points`` and it should work on windows as well
```suggestion
  script: "{{ PYTHON }} -m pip install . -vv"
  entry_points:
    - e3sm_to_cmip = e3sm_to_cmip.__main__:main
```

File: recipes/e3sm_to_cmip/meta.yaml (hunk 2 above)
Comment (xylar) [reply recipes/e3sm_to_cmip/meta.yaml]: @sterlingbaldwin, you would know better than me.

File: recipes/e3sm_to_cmip/meta.yaml (hunk 4 above)
Comment (xylar) [reply recipes/e3sm_to_cmip/meta.yaml]: Oops, thanks!

Comment (marcelotrevisani) [reply recipes/e3sm_to_cmip/meta.yaml]: After that, I think you can remove that selector to skip ``win`` and you might selector  the python versions which you want to on host and run requirements
After all that you can turn this recipe to be ``noarch: python``
https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python

File: recipes/e3sm_to_cmip/meta.yaml (hunk 2 above)
Comment (xylar) [reply recipes/e3sm_to_cmip/meta.yaml]: Actually, I think I can answer this.  We have some subprocess calls that have not been generalized for Windows (i.e. we call `ncremap` and not `ncreamp.exe`, as would be required in Windows).  We also don't foresee anyone wanting to use this package on Windows systems -- it really only makes sense on HPC.

File: recipes/e3sm_to_cmip/meta.yaml (hunk 4 above)
Comment (xylar) [reply recipes/e3sm_to_cmip/meta.yaml]: We could do that but I don't think we want to give the misimpression that Windows is supported.

File: recipes/e3sm_to_cmip/meta.yaml (hunk 2 above)
Comment (sterlingbaldwin) [reply recipes/e3sm_to_cmip/meta.yaml]: I dont think anything will break on py<3.6, but Im not going to be actively maintaining it for old python verions.

Comment (xylar) [reply recipes/e3sm_to_cmip/meta.yaml]: If the feedstock supports py<36 but you're not maintaining it, the builds are bound to start breaking.  So I think we should keep that constraint.  What about Windows?

Comment (marcelotrevisani) [reply recipes/e3sm_to_cmip/meta.yaml]: > Actually, I think I can answer this. We have some subprocess calls that have not been generalized for Windows (i.e. we call `ncremap` and not `ncreamp.exe`, as would be required in Windows). We also don't foresee anyone wanting to use this package on Windows systems -- it really only makes sense on HPC.

That is fine then. It was just to double check! :)

File: recipes/e3sm_to_cmip/meta.yaml (hunk 3 above)
Comment (sterlingbaldwin) [reply recipes/e3sm_to_cmip/meta.yaml]: This is fine for the time being, running with the "--help" finds most of the issues when dependencies dont load properly.

**Comment by xylar** [comment]
Thanks @marcelotrevisani!  I do appreciate your suggestion of going noarch.  I generally prefer that.

**Pull request merged**